src/Symfony/Component/Security/Core/Authorization/AccessDecisionManager.php

```diff
@@ -62,17 +78,27 @@ public function decide(TokenInterface $token, array $attributes, mixed $object =
     }
 
     /**
-     * @return \Traversable<int, int>
+     * @return \Traversable<int, Vote>
      */
-    private function collectResults(TokenInterface $token, array $attributes, mixed $object): \Traversable
+    private function collectVotes(TokenInterface $token, array $attributes, mixed $object): \Traversable
     {
         foreach ($this->getVoters($attributes, $object) as $voter) {
-            $result = $voter->vote($token, $object, $attributes);
-            if (!\is_int($result) || !(self::VALID_VOTES[$result] ?? false)) {
-                throw new \LogicException(\sprintf('"%s::vote()" must return one of "%s" constants ("ACCESS_GRANTED", "ACCESS_DENIED" or "ACCESS_ABSTAIN"), "%s" returned.', get_debug_type($voter), VoterInterface::class, var_export($result, true)));
+            if (method_exists($voter, 'getVote')) {
+                yield $voter->getVote($token, $object, $attributes);
+            } else {
+                yield new Vote($voter->vote($token, $object, $attributes));
```

Comment:
same if/else concern

I'm trying to reduce the number of method_exists checks to the bare minimum